programs/server/Server.cpp

```diff
@@ -721,7 +721,7 @@ try
     CurrentMetrics::set(CurrentMetrics::Revision, ClickHouseRevision::getVersionRevision());
     CurrentMetrics::set(CurrentMetrics::VersionInteger, ClickHouseRevision::getVersionInteger());
 
-    Poco::ThreadPool server_pool(3, server_settings.max_connections);
+    auto server_pool = std::make_unique<Poco::ThreadPool>(3, server_settings.max_connections);
```

Comment:
how does this affect anything?
maybe previous failures were related to accidentally moving the server_pool?

Reply:
There are two method of `Poco::ThreadPool`: `join()` and `release()`. The first if called on `servers_pool.joinAll()`, the second is called in destructor of `servers_pool`. The server hung on shutdown unless we call `release()` before `GlobalThreadPool::shutdown()`. May be second solution would be to change `join()` method to do what `release()` does, but I tried to leave it consistent with how it was.